File: LAB 4/Python/Translation.py
import matplotlib.pyplot as plt
import numpy as np

ORIGINAL_POINTS = np.array([[5, 5], [8, 8], [5, 11], [2, 8], [5, 5]])
TX, TY = 12, 3

plt.figure()
plt.grid(True)
plt.axis('equal')
plt.title('2D Translation')
plt.xlabel('X-axis')
plt.ylabel('Y-axis')

plt.plot(ORIGINAL_POINTS[:, 0], ORIGINAL_POINTS[:, 1], 'b-', linewidth=2, label='Original')
plt.fill(ORIGINAL_POINTS[:, 0], ORIGINAL_POINTS[:, 1], 'b', alpha=0.3)

TRANSLATED_POINTS = ORIGINAL_POINTS + [TX, TY]
plt.plot(TRANSLATED_POINTS[:, 0], TRANSLATED_POINTS[:, 1], 'r-', linewidth=2, label='Translated')
plt.fill(TRANSLATED_POINTS[:, 0], TRANSLATED_POINTS[:, 1], 'r', alpha=0.3)

plt.legend()
plt.show()

File: LAB 4/MATLAB/Translation.m
clc; clear; close all;

ORIGINAL_POINTS = [5, 5; 8, 8; 5, 11; 2, 8; 5, 5];
TX = 12; TY = 3;

figure;
hold on;
grid on;
axis equal;
title('2D Translation');
xlabel('X-axis');
ylabel('Y-axis');

plot(ORIGINAL_POINTS(:, 1), ORIGINAL_POINTS(:, 2), 'b-', 'LineWidth', 2);
fill(ORIGINAL_POINTS(:, 1), ORIGINAL_POINTS(:, 2), 'b', 'FaceAlpha', 0.3);

TRANSLATED_POINTS = ORIGINAL_POINTS + [TX, TY];
plot(TRANSLATED_POINTS(:, 1), TRANSLATED_POINTS(:, 2), 'r-', 'LineWidth', 2);
fill(TRANSLATED_POINTS(:, 1), TRANSLATED_POINTS(:, 2), 'r', 'FaceAlpha', 0.3);

legend('Original', 'Translated');
